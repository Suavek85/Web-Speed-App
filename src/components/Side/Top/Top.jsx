import React from 'react'
import { useSelector } from "react-redux"

import LightBulbUI from '../../svgs/LightBulb/LightBulbUI'
import { FIRST_PAINT_HEADER, FIRST_PAINT_CONTENT, RESULT } from '../../../constants/content.js'
import styles from './Top.scss'

export default function Top() {

  const getDataState = useSelector(state => state.stateReducer.getData)

  return (
    <div className={ styles.blockWrapper } >
      <div className={ styles.contentWrapper }>
        <h1>
          { FIRST_PAINT_HEADER }
          <LightBulbUI />
        </h1>
        <p>
          { FIRST_PAINT_CONTENT }
        </p>
        <p>
          { RESULT }
          <span>
            { getDataState[0] }
          </span>
        </p>
      </div>
    </div>
  )
}
