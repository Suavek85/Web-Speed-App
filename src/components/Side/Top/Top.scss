@import '../../../scss/styles.scss';

.resultstop__wrapper {
    width: 100%;
    height: 50%;
    display: flex;
    flex-direction: column;
    background-color: $color-top-background;
}

.resultstop__content-wrapper {
    padding: 0px 30px;
    > h1 {
      color: lightgrey;
    }
    > p:nth-child(2) {
      color: lightgrey;
    }
    > p:nth-child(3) {
      color: ivory;
      span {
        color: ivory;
      }
    }
}

:export {
    blockWrapper: resultstop__wrapper;
    contentWrapper: resultstop__content-wrapper;
}
