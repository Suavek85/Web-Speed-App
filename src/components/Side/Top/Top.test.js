import React from 'react'
import * as redux from 'react-redux'
import { shallow } from 'enzyme'
import Top from './Top'

describe('Top snapshot', () => {

  it('should match the snapshot', () => {
    const mockedResponse = [0.67, 0.47, "https://www.bbc.com/", "SLOW", "AVERAGE", "FAST", 0.98]
    const spyUseSelector = jest.spyOn(redux, 'useSelector')
    spyUseSelector.mockReturnValue(mockedResponse)
    const wrapper = shallow(<Top />)
    expect(wrapper).toMatchSnapshot()
  })

})