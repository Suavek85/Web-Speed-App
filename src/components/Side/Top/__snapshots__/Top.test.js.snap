// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top snapshot should match the snapshot 1`] = `
<Section
  content={
    Object {
      "header": "DOM size",
      "mainContent": "A large DOM will increase memory usage, cause longer style calculations.",
      "result": Array [
        "Result: ",
      ],
    }
  }
  data={
    Array [
      0.67,
    ]
  }
  position="top"
/>
`;
